metricbeat/helper/prometheus/prometheus.go

```diff
@@ -213,7 +216,10 @@ func (p *prometheus) ReportProcessedMetrics(mapping *MetricsMapping, r mb.Report
 		return
 	}
 	for _, event := range events {
-		r.Event(mb.Event{MetricSetFields: event})
+		r.Event(mb.Event{
+			MetricSetFields: event,
+			Namespace:       mapping.Namespace,
+		})
```

Comment:
Nice. I think this should be moved to its own PR, with a note in the developers changelog.

Reply:
+1 on having a separate PR. I think @sayden will also be happy to see this.

Reply:
thanks, 
created https://github.com/elastic/beats/issues/11423
pushed https://github.com/elastic/beats/pull/11424